src/deluge/gui/views/automation_instrument_clip_view.cpp

```diff
@@ -0,0 +1,3138 @@
+
+		//int32_t thisSquareEndPos = getPosFromSquare(xStartNow + 1, xScroll, xZoom) - effectiveRowLength * whichRepeat;
+
+		for (int32_t xDisplay = xStartNow; xDisplay < xEndNow; xDisplay++) {
+			if (xDisplay != xStartNow) {
+				thisSquareStartPos = squareEndPos[xDisplay - xStartNow - 1];
+				//thisSquareEndPos = squareEndPos[xDisplay - xStartNow];
+			}
+			int32_t i = searchTerms[xDisplay - xStartNow];
+
+			ParamNode* node = modelStack->autoParam->nodes.getElement(i - 1); // Subtracting 1 to do "LESS"
+
+			uint8_t* pixel = image + xDisplay * 3;
+
+			int32_t currentValue = modelStack->autoParam->getValueAtPos(thisSquareStartPos, modelStack);
+			int32_t knobPos = modelStack->paramCollection->paramValueToKnobPos(currentValue, modelStack);
+
+			knobPos = knobPos + kKnobPosOffset;
+
+			if (knobPos != 0 && knobPos >= yDisplay * kParamValueIncrementForAutomationDisplay) {
+
+				// If Node starts somewhere within square, draw the blur colour
+				if (node && node->pos > thisSquareStartPos) {
+
+					memcpy(pixel, &rowBlurColour[yDisplay], 3);
+					occupancyMask[xDisplay] = 64;
+				}
+
+				// Or if Node starts exactly on square...
+				else if (node && node->pos == thisSquareStartPos) {
+					memcpy(pixel, &rowColour[yDisplay], 3);
+					occupancyMask[xDisplay] = 64;
+				}
+
+				// Draw wrapped notes
+				//	else if (node && node->pos < thisSquareEndPos) {
+				//		memcpy(pixel, &rowTailColour[yDisplay], 3);
+				//		occupancyMask[xDisplay] = 64;
+				//	}
+
+				//Draw wrapped notes
+				else if (!drawRepeats || whichRepeat) {
+					bool wrapping = (i == 0); // Subtracting 1 to do "LESS"
+					                          //	if (wrapping) {
+					//		int32_t lastNodeIndex = modelStack->autoParam->nodes.getNumElements() - 1;
+					//		node = modelStack->autoParam->nodes.getElement(lastNodeIndex);
+					//	}
+					int32_t nodeEnd = effectiveRowLength - 1;
+					if (wrapping) {
+						nodeEnd -= effectiveRowLength;
+					}
+					if (nodeEnd > thisSquareStartPos && allowNoteTails) {
+						memcpy(pixel, &rowTailColour[yDisplay], 3);
+						occupancyMask[xDisplay] = 64;
+					}
+				}
+			}
+		}
+
+		xStartNow = xEndNow;
+		whichRepeat++;
+		// TODO: if we're gonna repeat, we probably don't need to do the multi search again...
+	} while (
+	    xStartNow
+	    != xEnd); // This will only do another repeat if we'd modified xEndNow, which can only happen if drawRepeats
+	*/
+}
+
+//easter egg lol. it is rendered when you press the CV clip button as you can't use automation view there
+//it draws a cute heart and musical note
+void AutomationInstrumentClipView::renderLove(uint8_t* image, uint8_t occupancyMask[], int32_t yDisplay) {
+
+	for (int32_t xDisplay = 0; xDisplay < kDisplayWidth; xDisplay++) {
+
+		uint8_t* pixel = image + (xDisplay * 3);
+
+		if (love[xDisplay][yDisplay] == 0xFFFFFFFF) {
+
+			memcpy(pixel, &rowColour[yDisplay], 3);
+			occupancyMask[xDisplay] = 64;
+		}
+	}
+}
+
+//no change compared to the instrument clip view version
```

Comment:
so I've adjusted the code so that it calls renderSidebar (that way you don't have duplicated code).

Everywhere else in the codebase where automationInstrumentClipVIew.sideBar was being referenced, i replaced it with instrumentClipView.sideBar

there is still a requirement to have this function defined in the automation UI because there are UI->sideBar calls in a couple places, namely these ones:

load_song_ui.cpp
getRootUI()->renderSidebar

ui.cpp
thisUI->renderSidebar

Please let me know if you're good with this change. Cheers

Reply:
looks good to me now, thanks.